R/snapshot-auto.R

```diff
@@ -113,6 +114,21 @@ renv_snapshot_auto_update <- function(project = renv_project_get() ) {
 
 renv_snapshot_task <- function() {
 
+  # treat warnings as errors in this scope
+  renv_scope_options(warn = 2L)
+
+  # attempt automatic snapshot, but disable on failure
+  withCallingHandlers(
+    renv_snapshot_task_impl(),
+    error = function(cnd) {
```

Comment:
```suggestion
  # attempt automatic snapshot, but disable on failure
  tryCatch(
    renv_snapshot_task_impl(),
    warning = function(cnd) {
```

?

Probably still also want to capture errors

Reply:
I set `options(warn = 2L)` above so that warnings are converted into errors (since I figure I want to treat both warnings and errors the same way here)

Reply:
Yeah, I just think you're better of doing that directly, rather than relying on R to convert warnings to errors (which travels through a fairly complex code path).